### PR TITLE
UIU-683: Update Renewal Confirmation window UI

### PR DESCRIPTION
**Pull request by OleksandrAntonenko1**
# Purpose
To allow users to override failed renewals in certain cases.
# Additional
This PR is only to create override modal. There is no backend integration. Also this PR includes code refactor witch was done in scope of story  UIU-742. 
# Screenshots

### CODE REVIEW

File: src/components/Loans/OpenLoans/components/OpenLoansSubHeader/OpenLoansSubHeader.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+              onToggle={this.onDropdownClick}
+              open={toggleDropdownState}
+            >
+              <Button
+                data-role="toggle"
+                align="end"
+                bottomMargin0
+                aria-haspopup="true"
+              >
+              Select Columns
```
Comment (VictorSoroka1) [inline src/components/Loans/OpenLoans/components/OpenLoansSubHeader/OpenLoansSubHeader.js]: Should this label be hardcoded?

File: src/components/Loans/OpenLoans/components/Modals/Modals.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { ChangeDueDateDialog } from '@folio/stripes/smart-components';
+import { stripesShape } from '@folio/stripes/core';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import BulkRenewalDialog from '@folio/users/src/components/BulkRenewalDialog';
+
```
Comment (VictorSoroka1) [inline src/components/Loans/OpenLoans/components/Modals/Modals.js]: Remove excess line.

File: src/components/BulkRenewalDialog/BulkRenewInfo.js [hunk 3]
```diff
@@ -1,76 +1,177 @@
...
+      }
+
+      return rewritableLoans;
+    }, []);
+  }
+
+  isOverridePossible(errorMessage) {
+    for (const rewritableMessage of this.rewritableMessages) {
+      const stringErrorMessage = get(errorMessage, 'props.values.message.props.values.message', '');
+      if (stringErrorMessage.includes(rewritableMessage)) {
```
Comment (VictorSoroka1) [inline src/components/BulkRenewalDialog/BulkRenewInfo.js]: Please add empty line after const and before return.

File: src/components/BulkOverrideDialog/BulkOverrideLoansList.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+          'itemStatus': <FormattedMessage id="ui-users.loans.columns.itemStatus" />,
+          'currentDueDate': <FormattedMessage id="ui-users.loans.columns.dueDate" />,
+          'requestQueue': <FormattedMessage id="ui-users.loans.details.requests" />,
+          'barcode': <FormattedMessage id="ui-users.information.barcode" />,
+          'callNumber': <FormattedMessage id="ui-users.loans.details.callNumber" />,
+          'renewals': <FormattedMessage id="ui-users.loans.columns.renewals" />,
+          'loanPolicy': <FormattedMessage id="ui-users.loans.details.loanPolicy" />,
+        }}
+        formatter={{
+          'isChecked': loan => (
```
Comment (VictorSoroka1) [inline src/components/BulkOverrideDialog/BulkOverrideLoansList.js]: 'isChecked' -> isChecked. Remove quotes.

File: src/components/BulkOverrideDialog/BulkOverrideLoansList.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        columnMapping={{
+          'isChecked': (
+            <input
+              type="checkbox"
+              checked={allChecked}
+              name="check-all"
+              onChange={toggleAll}
+            />
+          ),
+          'renewalStatus': <FormattedMessage id="ui-users.brd.header.renewalStatus" />,
```
Comment (VictorSoroka1) [inline src/components/BulkOverrideDialog/BulkOverrideLoansList.js]: Remove quotes.

File: src/components/BulkOverrideDialog/BulkOverrideInfo.js [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+  };
+
+  isLoanChecked = (id) => {
+    const { checkedLoans } = this.state;
+
+    return (id in checkedLoans);
+  };
+
+  handleDateTimeChanged = (datetime) => {
+    if (datetime !== 'Invalid date') {
```
Comment (Godlevskyi) [inline src/components/BulkOverrideDialog/BulkOverrideInfo.js]: Please move hardcoded string to constant

File: src/components/BulkOverrideDialog/BulkOverrideInfo.js [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+          loanPoliciesRecords={this.loanPoliciesRecords}
+          isLoanChecked={this.isLoanChecked}
+          toggleAll={this.toggleAll}
+          toggleItem={this.toggleItem}
+        />
+        <Row>
+          <Col sm={5}>
+            <TextArea
+              label={<FormattedMessage id="ui-users.additionalInfo.label" />}
+              placeholder={formatMessage({ id: 'ui-users.additionalInfo.placeholder' })}
```
Comment (Godlevskyi) [inline src/components/BulkOverrideDialog/BulkOverrideInfo.js]: Pleas avoid injecting intl when possible.
In this particular case you can refactor it to the following:
<FormattedMessage id="ui-users.additionalInfo.placeholder">
  {placeholder => (
    <TextArea
      label={<FormattedMessage id="ui-users.additionalInfo.label" />}
      placeholder={placeholder}
      required
      onChange={this.handleAdditionalInfoChange}
    />
  )}
</FormattedMessage>

File: src/components/BulkRenewalDialog/BulkRenewInfo.js [hunk 8]
```diff
@@ -1,76 +1,178 @@
...
+    requestCounts: PropTypes.object.isRequired,
+    errorMessages: PropTypes.object.isRequired,
+    onCancel: PropTypes.func.isRequired,
+  };
 
   constructor(props) {
     super(props);
+
+    this.rewritableMessages = [
+      'loan has reached its maximum number of renewals',
```
[the remark this answers is not in the document]
Comment (OleksandrAntonenko1) [reply src/components/BulkRenewalDialog/BulkRenewInfo.js]: This messages should not be translated cause they are coming from the BE without translation. Currently we are using error message to check type of an error but in future this behavior will be changed.

**Pull request merged**